### PR TITLE
Fix: #3729 Missing robots.txt get redirected to WP

### LINKED ISSUE #3729: WordPress Template robots.txt
### Describe the feature or change in detail

Hi,

Could I suggest a minor edit for the default WordPress nginx template please?

Can we amend the following section:

```
location = /robots.txt {
        allow all;
        log_not_found off;
        access_log off;
    }
```

To:

```
   location = /robots.txt {
        try_files $uri $uri/ /index.php?$args;
        log_not_found off;
        access_log off;
    }
```
Without this change, WordPress sites that use a PHP-generated ‘robots.txt’ instead of a 'real' robots.txt file give a 404 error.

Cheers

### Would you like to sponsor this feature to have it implemented?

No

### PR DESCRIPTION
**Pull request by jaapmarcus**

### CODE REVIEW

**Comment by jaapmarcus** [comment]
Closes: #3729

File: install/deb/templates/web/nginx/php-fpm/wordpress.stpl [hunk 1]
```diff
@@ -30,9 +30,9 @@ server {
 	}
 
 	location = /robots.txt {
-		allow all;
-		log_not_found off;
-		access_log off;
+			try_files $uri $uri/ /index.php?$args;
+			log_not_found off;
+			access_log off;
```
Comment (myrevery) [inline install/deb/templates/web/nginx/php-fpm/wordpress.stpl]: Please fix the indentation. : )

File: install/deb/templates/web/nginx/php-fpm/wordpress.tpl [hunk 2]
```diff
@@ -21,9 +21,9 @@ server {
 	}
 
 	location = /robots.txt {
-		allow all;
-		log_not_found off;
-		access_log off;
+			try_files $uri $uri/ /index.php?$args;
+			log_not_found off;
+			access_log off;
```
Comment (myrevery) [inline install/deb/templates/web/nginx/php-fpm/wordpress.tpl]: Also here.

**Review by myrevery — CHANGES_REQUESTED** [review]

File: install/deb/templates/web/nginx/php-fpm/wordpress_mu_subdir.stpl [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	access_log  /var/log/nginx/domains/%domain%.bytes bytes;
+	error_log   /var/log/nginx/domains/%domain%.error.log error;
+
+	# Rewrite requests to /wp-.* on subdirectory installs.
+	if (!-e $request_filename) {
+		rewrite /wp-admin$ $scheme://$host$uri/ permanent;
+		rewrite ^/[_0-9a-zA-Z-]+(/wp-.*) $1 last;
+		rewrite ^/[_0-9a-zA-Z-]+(/.*\.php)$ $1 last;
+	}
+
```
Comment (myrevery) [inline install/deb/templates/web/nginx/php-fpm/wordpress_mu_subdir.stpl]: @hudokkow Suggested to move before "location /" for consistency with other default templates format/style.

Also NGINX's "if" as special location block, which also avoids priority conflicts with other functions.

File: install/deb/templates/web/nginx/php-fpm/wordpress_mu_subdir.tpl [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	access_log  /var/log/nginx/domains/%domain%.bytes bytes;
+	error_log   /var/log/nginx/domains/%domain%.error.log error;
+
+	# Rewrite requests to /wp-.* on subdirectory installs.
+	if (!-e $request_filename) {
+		rewrite /wp-admin$ $scheme://$host$uri/ permanent;
+		rewrite ^/[_0-9a-zA-Z-]+(/wp-.*) $1 last;
+		rewrite ^/[_0-9a-zA-Z-]+(/.*\.php)$ $1 last;
+	}
+
```
Comment (myrevery) [inline install/deb/templates/web/nginx/php-fpm/wordpress_mu_subdir.tpl:24]: Also here.

**Review by myrevery — CHANGES_REQUESTED** [review]

File: install/deb/templates/web/nginx/php-fpm/wordpress.stpl (hunk 1 above)
Comment (jaapmarcus) [reply install/deb/templates/web/nginx/php-fpm/wordpress.stpl]: Changed...

File: install/deb/templates/web/nginx/php-fpm/wordpress.tpl (hunk 2 above)
Comment (jaapmarcus) [reply install/deb/templates/web/nginx/php-fpm/wordpress.tpl]: Same here
+ RPM

File: install/deb/templates/web/nginx/php-fpm/wordpress_mu_subdir.stpl (hunk 3 above)
Comment (hudokkow) [reply install/deb/templates/web/nginx/php-fpm/wordpress_mu_subdir.stpl]: See you already did the changes in the other branch. Fine by me.

Also no need to worry with attribution. Squash away... ;-)

Thanks!

Comment (jaapmarcus) [reply install/deb/templates/web/nginx/php-fpm/wordpress_mu_subdir.stpl]: Will add a attribution in the Changelog :)

**Pull request merged**